onnxruntime/contrib_ops/rocm/bert/batched_gemm_softmax_gemm_permute_pipelines.cuh

```diff
@@ -244,6 +283,246 @@ struct GemmSoftmaxGemmPermuteGenericPipeline {
   }
 };
 
+namespace {
+template <typename T>
+struct DataTypeAdaptor {
+  using type = T;
+};
+
+template <>
+struct DataTypeAdaptor<half> {
+  using type = ck::half_t;
+};
+
+template <>
+struct DataTypeAdaptor<BFloat16> {
+  using type = ck::bhalf16_t;
+};
+
+}  // namespace
+
+template <typename T>
+class GemmSoftmaxGemmPermuteTunableOp : public tunable::TunableOp<GemmSoftmaxGemmPermuteParams<T>> {
+ public:
+  GemmSoftmaxGemmPermuteTunableOp();
+
+  inline static bool IsSupportedMaskType(const AttentionParameters* attn) {
+    switch (attn->mask_type) {
+      case MASK_NONE:
+      case MASK_2D_DUMMY:
+      case MASK_2D_KEY_PADDING:
+      case MASK_3D_ATTENTION:
+      case MASK_4D_MEGATRON:
+        return true;
+      default:
+        return false;
```

Comment:
Is there plan to support MASK_1D_KEY_SEQ_LEN? That's major format for BERT in CUDA.

Reply:
Not at the moment. It is not possible to simply customize the elementwise op to accept it. So we need to add an adaptor kernel to convert the mask first.